include/zephyr/net/prometheus/collector.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_PROMETHEUS_COLLECTOR_H_
+#define ZEPHYR_INCLUDE_PROMETHEUS_COLLECTOR_H_
+
+/**
+ * @file
+ *
+ * @brief Prometheus collector APIs.
+ *
+ * @defgroup prometheus_collector Prometheus Collector API
+ * @since 3.8.0
+ * @version 0.1.0
+ * @ingroup prometheus
```

Comment:
I think we still need to `@defgroup prometheus` group (`@ingroup networking`) so that the APIs are visible. 

Also note, that with the following approach, each `@defgroup` will end up as a separate entry in the API summary page:
https://docs.zephyrproject.org/latest/develop/api/overview.html

I'll leave the decision up to you, but IMO we could simply have a single `@defgroup` entry:
```
 * @defgroup prometheus Prometheus API
 * @since 3.8.0
 * @version 0.1.0
 * @ingroup networking
```
And then just use the following in the remaining headers:
```
 * @addtogroup prometheus
```

Reply:
Missed out documents. Agree with you and updated your suggestions. Defined into collector.h file and just added '@addtogroup' field to other files.